source/slang/core.meta.slang

```diff
@@ -1720,6 +1720,48 @@ T __attachToNativeRef(NativeRef<T> nativeVal)
 }
 
 //@public:
+__generic<T, let N:int>
+__magic_type(ArrayExpressionType)
+struct Array : IRWArray<T>
+{
+    __intrinsic_op($(kIROp_GetArrayLength))
+    int getCount();
+}
+
+//@public:
+/// @category misc_types
+/// Represents a static UTF-8 string literal of `N` bytes.
+/// @remarks An implicit null terminator is not included in the byte count `N`.
+__generic<int N>
+__magic_type(ShortStringType)
+__intrinsic_type($(kIROp_ShortStringType))
+struct ShortString : IArray<uint>
+{
+    /// Get the number of bytes in the string.
+    /// @return The number of bytes in the string.
+    /// @remarks Does not count the implicit null terminator.
+    [__unsafeForceInlineEarly]
+    int getCount()
+    {
+        return N;
+    }
+
+    /// Get the number of bytes in the string.
+    /// @return The number of bytes in the string.
+    /// @remarks Does not count the implicit null terminator.
+    property uint length
```

Comment:
The convention in Slang is to use `int` for all size/count values. We don't use unsigned ints.